engine/consensus/matching/core.go

```diff
@@ -934,6 +931,7 @@ func (c *Core) requestPendingReceipts() (int, error) {
 		knownResultForBlock[s.Seal.BlockID] = struct{}{}
 	}
```

Comment:
I think we should be OK actually without iterating over receipts, because if it's already in Execution Tree, it must also be in the storage. If not, then it must be a unfinalized block, which we don't want to fetch yet.